samples/subsys/usb/mass/src/main.c

```diff
@@ -1,119 +1,155 @@
+ * Copyright (c) 2019-2020 Nordic Semiconductor ASA
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
-/* Sample to put the device in USB mass storage mode backed on a 16k RAMDisk. */
-
 #include <zephyr.h>
 #include <logging/log.h>
 #include <usb/usb_device.h>
+#include <fs/fs.h>
 #include <stdio.h>
 
 LOG_MODULE_REGISTER(main);
 
 #if CONFIG_DISK_ACCESS_FLASH
 #include <storage/flash_map.h>
+#endif
+
 #if CONFIG_FAT_FILESYSTEM_ELM
-#include <fs/fs.h>
 #include <ff.h>
+#endif
 
-static FATFS fat_fs;
-#define FATFS_MNTP      "/NAND:"
-
-static struct fs_mount_t fs_mnt = {
-	.type = FS_FATFS,
-	.mnt_point = FATFS_MNTP,
-	.storage_dev = (void *)FLASH_AREA_ID(storage),
-	.fs_data = &fat_fs,
-};
-#elif CONFIG_FILE_SYSTEM_LITTLEFS
-#include <fs/fs.h>
+#if CONFIG_FILE_SYSTEM_LITTLEFS
 #include <fs/littlefs.h>
-
 FS_LITTLEFS_DECLARE_DEFAULT_CONFIG(storage);
-static struct fs_mount_t fs_mnt = {
-	.type = FS_LITTLEFS,
-	.fs_data = &storage,
-	.storage_dev = (void *)FLASH_AREA_ID(storage),
-	.mnt_point = "/lfs",
-};
-#else
-#error No recognized file system
-#endif  /* file system */
-#endif  /* CONFIG_DISK_ACCESS_FLASH */
+#endif
 
-static void setup_disk(void)
+static struct fs_mount_t fs_mnt;
+
+static int setup_flash(struct fs_mount_t *mnt)
 {
+	int rc = 0;
 #if CONFIG_DISK_ACCESS_FLASH
-	struct fs_mount_t *mp = &fs_mnt;
-	unsigned int id = (uintptr_t)mp->storage_dev;
-	int rc;
+	unsigned int id;
 	const struct flash_area *pfa;
 
+	mnt->storage_dev = (void *)FLASH_AREA_ID(storage);
+	id = (uintptr_t)mnt->storage_dev;
+
 	rc = flash_area_open(id, &pfa);
 	printk("Area %u at 0x%x on %s for %u bytes\n",
 	       id, (unsigned int)pfa->fa_off, pfa->fa_dev_name,
 	       (unsigned int)pfa->fa_size);
 
-	if (IS_ENABLED(CONFIG_APP_WIPE_STORAGE)) {
+	if (rc != 0 && IS_ENABLED(CONFIG_APP_WIPE_STORAGE)) {
 		printk("Erasing flash area ... ");
 		rc = flash_area_erase(pfa, 0, pfa->fa_size);
 		printk("%d\n", rc);
 	}
 
-	rc = fs_mount(&fs_mnt);
+	if (rc != 0) {
+		flash_area_close(pfa);
+	}
+#endif
+	return rc;
+}
+
+static int mount_app_fs(struct fs_mount_t *mnt)
+{
+	int rc;
+
+#if CONFIG_FAT_FILESYSTEM_ELM
+	static FATFS fat_fs;
+
+	mnt->type = FS_FATFS;
+	mnt->fs_data = &fat_fs;
+	if (IS_ENABLED(CONFIG_DISK_ACCESS_RAM)) {
+		mnt->mnt_point = "/RAM:";
+	} else {
+		mnt->mnt_point = "/NAND:";
+	}
+
+#elif CONFIG_FILE_SYSTEM_LITTLEFS
+	mnt->type = FS_LITTLEFS;
+	mnt->mnt_point = "/lfs";
+	mnt->fs_data = &storage;
+#endif
+	rc = fs_mount(mnt);
+
+	return rc;
+}
+
+static void setup_disk(void)
+{
+	struct fs_mount_t *mp = &fs_mnt;
+	int rc;
+
+	if (IS_ENABLED(CONFIG_DISK_ACCESS_FLASH)) {
+		rc = setup_flash(mp);
+		if (rc != 0) {
+			LOG_ERR("Failed to setup flash area");
+			return;
+		}
+	}
+
+	if (!IS_ENABLED(CONFIG_FILE_SYSTEM_LITTLEFS) &&
+	    !IS_ENABLED(CONFIG_FAT_FILESYSTEM_ELM)) {
+		LOG_INF("No file system selected");
+		return;
+	}
+
+	rc = mount_app_fs(mp);
+	if (rc != 0) {
+		LOG_ERR("Failed to mount filesystem");
```

Comment:
You got the `rc`, maybe print the code in error log.

Reply:
It is redundant because fs_mount() logs every error.